script/cibuild

```diff
@@ -35,25 +36,27 @@ def main():
 
   if 'TARGET_ARCH' in os.environ:
     args = ['-t', os.environ['TARGET_ARCH']]
-    return run_ci(args, skip_upload, use_sccache, os.environ['COMPONENT'])
+    return run_ci(args, skip_upload, use_sccache, build_tests ,os.environ['COMPONENT'])
 
   if sys.platform in ['win32', 'cygwin']:
-    return (run_ci(['-t', 'x64'], skip_upload, use_sccache) or
-            run_ci(['-t', 'ia32'], skip_upload, use_sccache))
+    return (run_ci(['-t', 'x64'], skip_upload, use_sccache, build_tests) or
+            run_ci(['-t', 'ia32'], skip_upload, use_sccache, build_tests))
   elif sys.platform == 'linux2':
-    return (run_ci(['-t', 'x64'], skip_upload, use_sccache) or
-            run_ci(['-t', 'ia32'], skip_upload, use_sccache) or
-            run_ci(['-t', 'arm'], skip_upload, use_sccache) or
-            run_ci(['-t', 'arm64'], skip_upload, use_sccache))
+    return (run_ci(['-t', 'x64'], skip_upload, use_sccache, build_tests) or
+            run_ci(['-t', 'ia32'], skip_upload, use_sccache, build_tests) or
+            run_ci(['-t', 'arm'], skip_upload, use_sccache, build_tests) or
+            run_ci(['-t', 'arm64'], skip_upload, use_sccache, build_tests))
   else:
-    return run_ci([], skip_upload, use_sccache)
+    return run_ci([], skip_upload, use_sccache, build_tests)
```

Comment:
It seems like everything here is identical except for the architecture lists.

It would be slightly better to eliminate the redundancy, e.g.

```python
if 'TARGET_ARCH' in os.environ:
  archs = [os.environ['TARGET_ARCH']]
if sys.platform in ['win32', 'cygwin']:
  archs = ['x64', 'ia32']
elif sys.platform == 'linux2':
  archs = ['x64', 'ia32', 'arm', 'arm64']
else
  archs = []

for arch in archs:
  if not run_ci(['-t', arch], skip_upload, use_sccache, build_tests)
    return False

return True
```

2. ...and actually, since `run_ci()` is making assumptions on the args being passed in (it requires arg[1] to be the arch), maybe we should change `run_ci()`'s first argument from `args` to `target_arch` to remove that guesswork, and have `run_ci` build the args list itself?